spec/request/letters_request_spec.rb

```diff
@@ -139,13 +127,73 @@
       it 'should return a not found response' do
         VCR.use_cassette('evss/letters/beneficiary_500') do
           get '/v0/letters/beneficiary', nil, auth_header
-          expect(response).to have_http_status(:ok)
-          expect(response).to match_response_schema('letter_beneficiary')
-          expect(Oj.load(response.body).dig('meta', 'status')).to eq(
-            Common::Client::ServiceStatus::RESPONSE_STATUS[:server_error]
-          )
+          expect(response).to have_http_status(:internal_server_error)
+          expect(response).to match_response_schema('letters_errors')
         end
       end
     end
   end
+
+  describe 'error handling' do
+    context 'with a letter generator service error' do
+      it 'should return a not found response' do
+        VCR.use_cassette('evss/letters/letters_letter_generator_service_error') do
```

Comment:
Other than letter generator service error we haven't seen these errors in the wild and depend on EVSS or a EVSS sub-system having invalid data. So for those they're created manually which I agree is not ideal. I'll see if EVSS can set up some scenarios in CI.